pymc_marketing/clv/models/gamma_gamma.py

```diff
@@ -13,21 +13,27 @@
 
 
 class BaseGammaGammaModel(CLVModel):
-    def _process_priors(self, p_prior, q_prior, v_prior):
-        if p_prior is None:
-            p_prior = pm.HalfFlat.dist()
-        else:
-            self._check_prior_ndim(p_prior)
-        if q_prior is None:
-            q_prior = pm.HalfFlat.dist()
-        else:
-            self._check_prior_ndim(q_prior)
-        if v_prior is None:
-            v_prior = pm.HalfFlat.dist()
-        else:
-            self._check_prior_ndim(v_prior)
-
-        return super()._process_priors(p_prior, q_prior, v_prior)
+    def __init__(
+        self,
+        data: pd.DataFrame,
+        model_config: Optional[Dict] = None,
+        sampler_config: Optional[Dict] = None,
+    ):
+        super().__init__(model_config, sampler_config)
+        self.data = data
+        self.p_prior = self.create_distribution_from_prior(
+            self.model_config["p_prior"]["dist"],
+            **self.model_config["p_prior"]["kwargs"],
+        )
+        self.q_prior = self.create_distribution_from_prior(
+            self.model_config["q_prior"]["dist"],
+            **self.model_config["q_prior"]["kwargs"],
+        )
+        self.v_prior = self.create_distribution_from_prior(
+            self.model_config["v_prior"]["dist"],
+            **self.model_config["v_prior"]["kwargs"],
+        )
+        self._process_priors(self.p_prior, self.q_prior, self.v_prior)
```

Comment:
We should be still be calling `_check_prior_ndim`, to make sure users are not trying to create multivariate priors or priors with the wrong shape, such as `{"p_prior": {"dist": "Normal", kwargs: {"mu": [0, 0, 0]}}`.

By the way, how does serialization handle numpy objects in the kwargs?

Reply:
so far it would need to be added manually, that's where the serialisable model config comes in. I'll create an users guide to that initially, with time I'd say we should aim for having a module listing most common non-serialisable structures and providing their serializable equivalents for the purpose of saving

Reply:
I could add a function like that:
`def make_serializable(obj):
    """
    Make an object json serializable by converting non-serializable fields to strings.
    """
    # Base case: if obj is serializable, we're done.
    try:
        json.dumps(obj)
        return obj
    except TypeError:
        pass

    # If obj is a dictionary, recursively make its fields serializable.
    if isinstance(obj, dict):
        return {k: make_serializable(v) for k, v in obj.items()}

    # If obj is a list, recursively make its elements serializable.
    elif isinstance(obj, list):
        return [make_serializable(elem) for elem in obj]

    # If obj is a numpy array, convert it to a list.
    elif isinstance(obj, np.ndarray):
        return obj.tolist()

    # If obj is a datetime, convert it to a string.
    elif isinstance(obj, datetime.datetime):
        return obj.isoformat()

    # If obj is a non-serializable object, convert it to a string.
    else:
        return str(obj)`

and then our serialisation would look like that:
I would add a setter for the _serializable_model_config so users can manually add cases that are not yet covered by the make_serializable function
`@property
def _serializable_model_config(self):
    return self.make_serializable(self.model_config)
`
what do you think about it?

Reply:
> We should be still be calling `_check_prior_ndim` (in all models), to make sure users are not trying to create multivariate priors or priors with the wrong shape, such as `{"p_prior": {"dist": "Normal", kwargs: {"mu": [0, 0, 0]}}`.
> 
> By the way, how does serialization handle numpy objects in the kwargs?

I've included _check_prior_ndim inside create_distribution_for_prior, this way current code can stay as-is (maybe with small tweak regarding verbosity as you requested) and the check will be done

Reply:
@ricardoV94 the prior creation was adapted and renamed as you requested, please in a free moment provide feedback on the serialisation concept proposed above